src/main/scala/exercise2/Queue.scala

```diff
@@ -0,0 +1,34 @@
+package exercise2
+
+import scala.util.Try
+
+trait QueueLike[T] {
+  def enqueue(elem: T): QueueLike[T]
+
+  def dequeue(): Try[QueueLike[T]]
+
+  def front(): Option[T]
+
+  def isEmpty: Boolean
+}
+
+class Queue[T](
+    val in: StackLike[T] = new Stack[T](),
+    val out: StackLike[T] = new Stack[T]()
+) extends QueueLike[T] {
+
+  def enqueue(elem: T): QueueLike[T] = if (isEmpty)
+    new Queue(in, out.push(elem))
+  else new Queue(in.push(elem), out)
+
+  def dequeue(): Try[QueueLike[T]] = {
+    if (isEmpty) throw new Throwable("Queue is empty")
```

Comment:
Very bad! Why bother with `Try` when you throw an exception here anyway?
instead use
```scala
new Failure(new Throwable("Queue is empty"))
```